GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitRepoTests.cs

```diff
@@ -44,6 +44,7 @@ public abstract class GitRepoTests
             "foo.cpp", // Changes to a folder in one test
             "FilenameEncoding",
             "GitCommandsTests",
+            "GVFLT_MultiThreadTest",

```

Comment:
Why was this folder added to the list?

Reply:
That allows the `TestValidationAndCleanup()` method to succeed on case-sensitive systems, because as memory serves, [`DeleteFolderAndChangeBranchToFolderWithDifferentCase()`](https://github.com/microsoft/VFSForGit/blob/e534dfc518249044b5edbb6de060a323310e8a0b/GVFS/GVFS.FunctionalTests/Tests/GitCommands/CheckoutTests.cs#L742) leaves behind a top-level folder of that name, and the test cleanup only succeeds if it matches.

Specifically, [`TestValidationAndCleanup()`](https://github.com/microsoft/VFSForGit/blob/e534dfc518249044b5edbb6de060a323310e8a0b/GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitRepoTests.cs#L184) -- when running in sparse mode only -- passes the list of `SparseModeFolders` to `WithDeepStructure()`.  In this PR we also adjust